File: example/src/main/java/com/example/rxactivityexample/MainActivity.java
package com.example.rxactivityexample;

import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.widget.Button;

import butterknife.BindView;

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
public class MainActivity extends BaseActivity {

    @BindView(R.id.toolbar) Toolbar toolbar;
    @BindView(R.id.button1) Button button1;
    @BindView(R.id.button2) Button button2;

    @Override
    protected int getContentView() {
        return R.layout.activity_main;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setSupportActionBar(toolbar);
    }
}

File: example/src/com/example/rxactivityexample/MainActivity.kt
package com.example.rxactivityexample

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.support.design.widget.errorbar
import android.support.v7.app.AppCompatActivity
import com.hendraanggrian.rx.activity.*
import io.reactivex.rxkotlin.subscribeBy
import kotlinx.android.synthetic.main.activity_main.*

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)

        val intent = Intent(this, NextActivity::class.java)
        button1.setOnClickListener {
            startActivityForOk(intent)
                    .subscribeBy(
                            onNext = { _ -> errorbar(coordinatorLayout, "onNext", android.R.string.ok, {}) },
                            onError = { e -> errorbar(coordinatorLayout, "onError: ${e.message}", android.R.string.ok, {}) })
        }
        button2.setOnClickListener {
            startActivityForAny(intent)
                    .subscribeBy(
                            onNext = { result ->
                                val print = "ActivityResult[requestCode=${result.getRequestCode()}, resultCode=${when (result.getResultCode()) {
                                    Activity.RESULT_OK -> "RESULT_OK"
                                    Activity.RESULT_CANCELED -> "RESULT_CANCELED"
                                    else -> result.getResultCode().toString()
                                }}]"
                                errorbar(coordinatorLayout, "onNext:\n$print", android.R.string.ok, {})
                            },
                            onError = { e -> errorbar(coordinatorLayout, "onError: ${e.message}", android.R.string.ok, {}) })
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        onActivityResultBy(requestCode, resultCode, data)
    }
}